### PR TITLE
[dom-mc] jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb with Julia[Extensions]

### PR DESCRIPTION
**Pull request by twrobinson**
This requires JuliaExtensions/1.4.2-CrayGNU-20.06 (help wanted, Sam)

### CODE REVIEW

**Comment by twrobinson** [comment]
Awaiting JuliaExtensions/1.4.2-CrayGNU-20.06

**Comment by lucamar** [comment]
The pull request #1786 will provide `JupyterLab` with `Julia 1.5.0`: should we close this pull request @omlins?

**Comment by omlins** [comment]
> he pull request #1786 will provide `JupyterLab` with `Julia 1.5.0`: should we close this pull request @omlins?

No, we can keep this open. I will update it to use Julia 1.5.0.

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+easyblock = 'PythonBundle'
+
+name = 'jupyterlab'
+version = '1.2.16'
+versionsuffix = '-batchspawner'
+
+_pyminver = '6'
+
+_jl_maj_ver = '1'
+_jl_min_ver = '4'
```
Comment (omlins) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb:12]:
```suggestion
_jl_min_ver = '5'
```

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+_jl_rev_ver = '2'
```
Comment (omlins) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb:13]:
```suggestion
_jl_rev_ver = '0'
```

**Comment by omlins** [comment]
I added Julia and JuliaExtensions for PE 20.06 in here.

**Comment by lucamar** [comment]
The build of `jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb` failed as in #1813 while installing the extension `ipyparaview`: 
```
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@jupyterlab%2fservices - N
ot found
```
It seems that the download from the web site is problematic today, I will try to test it again as @henrique did successfully for #1813.

See https://jenkins.cscs.ch/blue/organizations/jenkins/TestingEB/detail/TestingEB/3943/pipeline and the log file at `/scratch/snx3000tds/jenscscs/TestingEB/dom-mc/82c4df1-3943/tmp/eb-wYMekw/easybuild-jupyterlab-1.2.16-20200806.112938.SDthJ.log`.

**Comment by lucamar** [comment]
retest this please

**Comment by lucamar** [comment]
Now the build of `jupyterlab-1.2.16-CrayGNU-20.06-batchspawner.eb` failed while installing the extension `bash_kernel 0.7.2`, with the error ```
ERROR: Exception: raise BackendUnavailable
pip._vendor.pep517.wrappers.BackendUnavailable
```
The issue seems to occur with `pip 19` in other cases as well: see https://github.com/pypa/pip/issues/6164
By the way, since we have so many extensions in this recipe, we could also upgrade `pip`:
``` 
WARNING: You are using pip version 19.2.3, however version 20.2.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```
The full output is available at this link:
https://jenkins.cscs.ch/blue/organizations/jenkins/TestingEB/detail/TestingEB/3949/pipeline

The log file with the error is`/scratch/snx3000tds/jenscscs/TestingEB/dom-mc/fd611d2-3949/tmp/eb-1WE1y6/easybuild-jupyterlab-1.2.16-20200806.155248.vQNrk.log`

**Comment by omlins** [comment]
retest this please

**Comment by lucamar** [comment]
> retest this please
Unfortunately, it is still failing while installing the extension `bash_kernel 0.7.2`, with the same error as before. 
I think that we should try a different `pip` version, as they did in https://github.com/pypa/pip/issues/6164.

**Comment by lucamar** [comment]
The issue with `bash_kernel 0.7.2` is reported also here: https://github.com/takluyver/bash_kernel/issues/100

**Comment by omlins** [comment]
retest this please

**Comment by lucamar** [comment]
I have added `'use_pip': False` for the extension `bash_kernel` as well, since it seems to succeed for other extensions.

**Comment by omlins** [comment]
> I have added `'use_pip': False` for the extension `bash_kernel` as well, since it seems to succeed for other extensions.

Great, @twrobinson is that OK?

**Pull request merged**